capstone/cite/tests/test_views.py

```diff
@@ -149,3 +157,75 @@ def test_single_case(client, auth_client, django_assert_num_queries, case):
     check_response(response, content_includes=case_text)
     auth_client.auth_user.refresh_from_db()
     assert auth_client.auth_user.case_allowance_remaining == settings.API_CASE_DAILY_ALLOWANCE - 1
+
+
+@pytest.mark.django_db
+def test_schema_in_case(client, case):
+    # setup
+    url = case.get_frontend_url()
+    parsed = parse_xml(case.case_xml.orig_xml)
+    case_text = parsed('casebody|casebody').children()[10].text.replace('\xad', '')
+
+    ### whitelisted case
+
+    case.jurisdiction.whitelisted = True
+    case.jurisdiction.save()
+
+    response = client.get(url)
+    check_response(response, content_includes=case_text)
+
+    schema = get_schema(response)
+    assert schema["headline"] == case.name_abbreviation
+    assert schema["author"]["name"] == case.court.name
+
+    # if case is whitelisted, extra info about inaccessibility is not needed
+    # https://developers.google.com/search/docs/data-types/paywalled-content
+    assert "hasPart" not in schema
+
+    ### blacklisted case
+
+    case.jurisdiction.whitelisted = False
+    case.jurisdiction.save()
+
+    response = client.post(reverse('set_cookie'), {'not_a_bot': 'yes', 'next': url}, follow=True)
+    check_response(response, content_includes=case_text)
+
+    schema = get_schema(response)
+    assert schema["headline"] == case.name_abbreviation
+    assert schema["author"]["name"] == case.court.name
+
+    # if case is blacklisted, we include more data
+    assert "hasPart" in schema
+    assert schema["hasPart"]["isAccessibleForFree"] == 'False'
+
+
+@pytest.mark.django_db()
+def test_schema_in_case_as_google_bot(client, case, monkeypatch):
+    # setup
+    url = case.get_frontend_url()
+    parsed = parse_xml(case.case_xml.orig_xml)
+    case_text = parsed('casebody|casebody').children()[10].text.replace('\xad', '')
+
+    def mock_is_google_bot(request):
+        return True
+
+    monkeypatch.setattr(helpers, 'is_google_bot', mock_is_google_bot)
+
+    # our bot has seen too many cases!
+    session = client.session
+    session['case_allowance_remaining'] = 0
+    session.save()
+    assert session['case_allowance_remaining'] == 0
+
+    case.jurisdiction.whitelisted = False
+    case.jurisdiction.save()
+
+    response = client.get(url, follow=True)
```

Comment:
Let's add an `assert not is_cached(response)` in here to make sure the response isn't getting cached.